module/trie.c

```diff
@@ -0,0 +1,271 @@
+/* This file implements a new module native data type called "TRIE".
+ * The data structure is based on the idea of a trie (aka a prefix 
+ * tree) and largely works the same way.
+ */
+#define IN_TRIE 1
+#define NOT_IN_TRIE 0 
+#define PARTIAL_IN_TRIE (-1)
+
+#include "../redismodule.h"
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+#include <math.h>
+
+
+static RedisModuleType *trie;
+
+/* ========================== Internal data structure (Bare bones functions)  ======================= */
+
+struct trie {
+    char current; 
+        // The first trie_t will be '/0' for any Trie.
+    struct trie **children;
+         // ALPHABET_SIZE is 256 for all possible characters.
+    int is_word; 
+        // if is_word is 1, indicates that this is the end of a word. Otherwise 0.
+    struct trie *parent;
+        // parent trie for traversing backwards
+};
+
+/*
+    Creates and allocates memory for new trie.
+    
+    Parameters:
+     - current: A char for the current character
+    
+    Returns:
+     - A pointer to the trie, or REDISMODULE_ERR if a pointer 
+       cannot be allocated
+*/
+struct trie *trie_new (char current)
+{
+    struct trie *t = RedisModule_Calloc(1,sizeof(struct trie));
+    if (t == NULL){
```

Comment:
Please use spaces here - smashing symbols together like this is pretty ugly. There are many occurrences of this throughout the file, please fix them all.